service/src/main/java/org/mskcc/smile/service/impl/SampleServiceImpl.java

```diff
@@ -32,6 +32,7 @@
 @Component
 public class SampleServiceImpl implements SmileSampleService {
     private static final Pattern DMP_PATIENT_ID = Pattern.compile("P-\\d*");
+    private static final Pattern RECAPTURE_INVESTIGATOR_SAMPLE_ID = Pattern.compile("^P-\\d{6}-.*-WES$");
```

Comment:
Per discussion with Ben and Pav, a sample is a recapture if the investigator sample ID (e.g. P-123456-T01-WES) starts with the DMP Patient ID (i.e. "P-123456") and ends with "WES".